### PR TITLE
Fix Resource Paths in REPL

### LINKED ISSUE #267: VFS can't find files without packages
[Ed.: This issue has had a few incarnations now. But basically VFS can fail to resolve assets at the root of the VFS under some REPLs. This is a bug.]

----

The following error was obtained when I ran the code in iPython without an image file present. 
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-1-eca4d86b41c8> in <module>()
     12
     13
---> 14 ppb.run(setup=setup)

~\AppData\Local\Continuum\anaconda3\lib\site-packages\ppb\__init__.py in run(setup, log_level, starting_scene)
     32
     33     with GameEngine(starting_scene, **kwargs) as eng:
---> 34         eng.run()

~\AppData\Local\Continuum\anaconda3\lib\site-packages\ppb\engine.py in run(self)
     77         else:
     78             self.start()
---> 79             self.main_loop()
     80
     81     def start(self):

~\AppData\Local\Continuum\anaconda3\lib\site-packages\ppb\engine.py in main_loop(self)
     91                     self.signal(event)
     92                     while self.events:
---> 93                         self.publish()
     94             self.manage_scene()
     95

~\AppData\Local\Continuum\anaconda3\lib\site-packages\ppb\engine.py in publish(self)
    113         self.__event__(event, self.signal)
    114         for system in self.systems:
--> 115             system.__event__(event, self.signal)
    116         # Required for if we publish with no current scene.
    117         # Should only happen when the last scene stops via event.

~\AppData\Local\Continuum\anaconda3\lib\site-packages\ppb\events.py in __event__(self, bag, fire_event)
     74             try:
     75                 elog.debug(f"Calling handler {meth} for {name}")
---> 76                 meth(bag, fire_event)
     77             except TypeError as ex:
     78                 from inspect import signature

~\AppData\Local\Continuum\anaconda3\lib\site-packages\ppb\systems\__init__.py in on_render(self, render_event, signal)
     61
     62         for game_object in render_event.scene:
---> 63             resource = self.prepare_resource(game_object)
     64             if resource is None:
     65                 continue

~\AppData\Local\Continuum\anaconda3\lib\site-packages\ppb\systems\__init__.py in prepare_resource(self, game_object)
     76             return None
     77         if image_name not in self.resources:
---> 78             self.register_renderable(game_object)
     79
     80         source_image = self.resources[game_object.image]

~\AppData\Local\Continuum\anaconda3\lib\site-packages\ppb\systems\__init__.py in register_renderable(self, renderable)
    105     def register_renderable(self, renderable):
    106         image_name = renderable.__image__()
--> 107         source_path = renderable.__resource_path__()
    108         self.register(source_path / image_name, image_name)
    109

~\AppData\Local\Continuum\anaconda3\lib\site-packages\ppb\sprites.py in __resource_path__(self)
    237     def __resource_path__(self):
    238         if self.resource_path is None:
--> 239             self.resource_path = Path(realpath(getfile(type(self)))).absolute().parent
    240         return self.resource_path

~\AppData\Local\Continuum\anaconda3\lib\inspect.py in getfile(object)
    651             if getattr(object, '__file__', None):
    652                 return object.__file__
--> 653         raise TypeError('{!r} is a built-in class'.format(object))
    654     if ismethod(object):
    655         object = object.__func__

TypeError: <module '__main__'> is a built-in class

```

### PR DESCRIPTION
**Pull request by astronouth7303**
Fix the crashing bug when running PPB in a REPL by using the CWD when the module is `__main__`.

Closes #267.

### CODE REVIEW

**Comment by astronouth7303** [comment]
Tests incoming.

File: tests/test_sprites.py [hunk 1]
```diff
@@ -325,3 +325,14 @@ def test_rotatable_base_sprite():
 
     test_sprite.rotate(1)
     assert test_sprite.rotation == 1
+
+
+def test_sprite_in_main():
```
Comment (pathunstrom) [inline tests/test_sprites.py]: This doesn't fail if you revert the change. Doing some experimentation to find a way to force it to fail.

File: ppb/sprites.py [hunk 2]
```diff
@@ -296,5 +293,8 @@ def __image__(self):
 
     def __resource_path__(self):
         if self.resource_path is None:
-            self.resource_path = Path(realpath(getfile(type(self)))).absolute().parent
+            if type(self).__module__ == '__main__':
```
Comment (pathunstrom) [inline ppb/sprites.py]: This breaks the examples if you run them from the project root.

**Comment by pathunstrom** [comment]
Okay, as per slack, going to merge this.
